plugin.video.arloview/resources/lib/arlo_stream.py

```diff
@@ -160,29 +145,66 @@ def _get_camera_info(self, camera):
         return fmt_str.format(camera['deviceId'],
                               camera['properties']['modelId'],
                               camera['batteryLevel'],
-                              camera['signalStrength'])
+                              camera['signalStrength']) 
 
+    
+    
     def _get_camera_snapshot(self, device_id):
-        if REAL_SETTINGS.getSetting('show_snapshots'):
-            camera = self._get_camera(device_id)
-            snapshot_file = "{0}/resources/media/{1}.jpg".format(ADDON_PATH, camera['deviceId'])
-            snapshot_url = camera['presignedFullFrameSnapshotUrl']
-            #self.log("*** file: {}".format(snapshot_file))
-            #self.log("*** URL : {}".format(snapshot_url))
+        if not REAL_SETTINGS.getSettingBool('show_snapshots'):
+            snapshot_file = "{0}/resources/ArloCamera.png".format(ADDON_PATH)
+        else:
+            snapshot_path = "{0}/resources/media".format(SETTINGS_LOC)
+            if not os.path.exists(snapshot_path):
+                os.makedirs(snapshot_path)
+            snapshot_file = self._refresh_snapshot(device_id, snapshot_path)
+
+        self.log("  snapshot_file name = {0}".format(snapshot_file), xbmc.LOGDEBUG)
+        return snapshot_file
+
+    def _refresh_snapshot(self, device_id, snapshot_path):
+        camera = self._get_camera(device_id)
+        snapshot_type = REAL_SETTINGS.getSetting('snapshot_type')
+        snapshot_file_prefix = ''.join(ch for ch in snapshot_type if ch.isupper())
+        snapshot_file = "{0}/{1}_{2}.jpg".format(snapshot_path, snapshot_file_prefix, camera['deviceId'])
+        if self._snapshot_expired(snapshot_file):
+            snapshot_url = camera[snapshot_type]
             try:
+                self.log("Download {} into {}_{}".format(snapshot_type, snapshot_file_prefix, camera['deviceId']))
                 self.arlo.DownloadSnapshot(snapshot_url, snapshot_file, 4096)
-            except:
-                self.log("Unable to download snapshot: {0}".format(snapshot_url), xbmc.LOGERROR)
+                if os.path.getsize(snapshot_file) < 1024:
+                    # File seems to be invalid, remove so it will try again the next time
+                    self.log("Hmm, seems to be an invalid jpg", xbmc.LOGERROR)
+                    self.log("  FILE: {}".format(snapshot_file), xbmc.LOGERROR)
+                    snapshot_file = "{0}/resources/ArloCamera.png".format(ADDON_PATH)
+            except Exception as err:
```

Comment:
Log spam

Reply:
Hmm.  These messages only occur when there is an error during processing.  If a user encounters these conditions, without a message, I will have no visibility as to what is going on.

I've changed the self.log() method, to ONLY output to the log when addon DEBUG is turned on (in settings), as below.  Will that suffice to limit the log spam?

        if self.debug_mode :
            # If the plugin debug is turned on, allow the message to print (NOTICE)
            level = xbmc.LOGNOTICE
            xbmc.log("[{}-{}] {}".format(ADDON_ID, ADDON_VERSION, msg), level)